### PR TITLE
sd-bus: Add sd_bus_message_read_strv docs

### PR DESCRIPTION
**Pull request by daandemeyer**

### CODE REVIEW

File: man/sd_bus_message_read_strv.xml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      </funcprototype>
+    </funcsynopsis>
+  </refsynopsisdiv>
+
+  <refsect1>
+    <title>Description</title>
+
+    <para><function>sd_bus_message_read_strv()</function> gives access to an array of strings in message
+    <parameter>m</parameter>. The "read pointer" in the message must be right before an array of strings. A
+    pointer to the <constant>NUL</constant>-terminated array of strings is returned in the output parameter
```
Comment (keszybz) [inline man/sd_bus_message_read_strv.xml]: Ha, gotcha. NUL is used to refer to a byte, and NULL to a pointer. So it's NULL-terminated array of NUL-terminated strings in this case. So please change that to NULL.

File: man/sd_bus_message_read_strv.xml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    <parameter>l</parameter>. Note that ownership of any string array stored in <parameter>l</parameter> by
+    <function>sd_bus_message_read_strv()</function> is transferred to the caller. Hence, the caller is
+    responsible for freeing the results returned by this function.</para>
```
Comment (keszybz) [inline man/sd_bus_message_read_strv.xml]: I think this should be clarified whether the array itself, or the array including the strings must be freed by the caller. (It's the latter.)

File: man/sd_bus_message_read_strv.xml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  </refsect1>
+
+  <refsect1>
+    <title>Return Value</title>
+
+    <para> On success, <function>sd_bus_message_read_strv()</function> returns a non-negative integer. On
```
Comment (keszybz) [inline man/sd_bus_message_read_strv.xml]: Extra space here.

File: man/sd_bus_message_read_strv.xml [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    failure, it returns a negative errno-style error code.</para>
```
Comment (keszybz) [inline man/sd_bus_message_read_strv.xml:50]: I think we should make sd_bus_message_read_strv return the length of `l` at some point...

**Comment by DaanDeMeyer** [comment]
I clarified freeing the array and its contents stuff for `sd_bus_list_names` as well

File: man/sd_bus_list_names.xml [hunk 5]
```diff
@@ -46,8 +46,8 @@
     <ulink url="https://dbus.freedesktop.org/doc/dbus-specification.html#bus-messages-list-activatable-names">
     org.freedesktop.DBus.ListActivableNames</ulink> to retrieve the list of all names on the bus that can be
     activated. Note that ownership of any string arrays returned by <function>sd_bus_list_names()</function>
-    is transferred to the caller and hence, the caller is responsible for freeing any results stored by
-    <function>sd_bus_list_names()</function> in <parameter>acquired</parameter> and
+    is transferred to the caller and hence, the caller is responsible for freeing any arrays (and its
+    contents) stored by <function>sd_bus_list_names()</function> in <parameter>acquired</parameter> and
```
Comment (keszybz) [inline man/sd_bus_list_names.xml]: *their* contents

File: man/sd_bus_message_read_strv.xml [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    </funcsynopsis>
+  </refsynopsisdiv>
+
+  <refsect1>
+    <title>Description</title>
+
+    <para><function>sd_bus_message_read_strv()</function> gives access to an array of strings in message
+    <parameter>m</parameter>. The "read pointer" in the message must be right before an array of strings. A
+    pointer to the <constant>NULL</constant>-terminated array strings is returned in the output parameter
+    <parameter>l</parameter>. Note that ownership of any string array stored in <parameter>l</parameter> by
```
Comment (keszybz) [inline man/sd_bus_message_read_strv.xml]: of any string array → of the array ?

File: man/sd_bus_message_read_strv.xml [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+      </funcprototype>
+    </funcsynopsis>
+  </refsynopsisdiv>
+
+  <refsect1>
+    <title>Description</title>
+
+    <para><function>sd_bus_message_read_strv()</function> gives access to an array of strings in message
+    <parameter>m</parameter>. The "read pointer" in the message must be right before an array of strings. A
+    pointer to the <constant>NULL</constant>-terminated array strings is returned in the output parameter
```
Comment (keszybz) [inline man/sd_bus_message_read_strv.xml]: array *of* strings

**Comment by DaanDeMeyer** [comment]
I ended up modifying the entire paragraph. PTAL

**Pull request merged**